### DISCUSSION TITLE
Change default path for EGS Launcher on Unix.

### DISCUSSION THREAD

**Discussion by vkbsb** [PhilipK/BoilR#300] [General]
Currently the EGS default launcher path picks EpicGamesLauncher.exe from Engine/Binaries/Win32/. Currently when i try to launch games its just launching the EGS but not the game. It works as expected when we pick the binary from Portal/Binaries/Win64/ directory instead. Would like to get this fixed if you are ok with it @PhilipK

**Comment by PhilipK** [comment]
Yeah sure, if you make a PR I will review it

**Reply by vkbsb** [reply]
https://github.com/PhilipK/BoilR/pull/301

**Reply by vkbsb** [reply]
I just realized that the Win32 executable still just launches the EGS but not the game, I needed to use the Win64/ executable and also had to force the proton compatibility. How is the proton compatibility picked right now?

**Reply by PhilipK** [reply]
It seems pretty flaky if steam picks up on the proton version, but i try to set it here:
https://github.com/PhilipK/BoilR/blob/main/src/steam/proton_vdf_util.rs#L17

Trying to add some strings to a vdf file

**Reply by vkbsb** [reply]
@PhilipK  is there a reason to hold on to the Binaries/Win32 directory for the EGS? on my SD only the 64 Bit version is directly launching the game, Could be because i've downloaded them through the 64bit version.

**Reply by vkbsb** [reply]
i mean can i switch the default EpicGamesLauncher.exe to be picked from Win64 instead of the Win32 directory?

**Reply by PhilipK** [reply]
How about you make a pr where you make the switch, then I can test it on my SD and my laptop, then we make sure it works on more than one machine.

**Reply by moraroy** [reply]
Hey guys, I noticed Boilr got updated, good work! I believe you are missing a hyphen to launch the games from Epic Launcher. Like this...  -'com.epicgames... it should work for both Win32 and Win64. Everything else looks good on my end for Importing Epic Games.

**Reply by PhilipK** [reply]
@moraroy  so if we just change this line:
https://github.com/PhilipK/BoilR/blob/160442b25fac9b00ac13e7ec5a9d4504be4fc993/src/platforms/egs/manifest_item.rs#L85

So that it goes from:
STEAM_COMPAT_DATA_PATH="folder_name_here" %command% 'epic-url-here'

To:
So that it goes from:
STEAM_COMPAT_DATA_PATH="folder_name_here" %command% -'epic-url-here'

Then it should work?

**Reply by moraroy** [reply]
I promise it will.

**Reply by PhilipK** [reply]
https://github.com/PhilipK/BoilR/pull/304

**Reply by PhilipK** [reply]
https://github.com/PhilipK/BoilR/releases/tag/v.1.7.17

Pre-release version ready for people to test

**Reply by vkbsb** [reply]
Can confirm it works.  
It's Strange because i've got a couple of shortcuts set without the '-' and they work just fine with the Win64/EpicGameLauncher.exe..

**Reply by vkbsb** [reply]
@PhilipK  how do i get boiler to update the existing links should i be disconnecting them and importing them again?

**Reply by PhilipK** [reply]
@vkbsb congratualtions I think you just found another bug.
To solve it for now, remove them, run import, then add them and run import again.

The problem is that steam createa a unique id for each shortcut, called app id.
It is a weird hash of the exe location and the shortcut name, but it does not include the launch options.

Now when boilr checks what to do when it should import, it only checks those app ids as well. This means that the launcher parameter does not get updated.

**Reply by moraroy** [reply]
I have some other fixes for BoilR I'll share tomorrow. Good work brothers.

**Reply by vkbsb** [reply]
@PhilipK  so im' thinking we should be doing the following for the update launch options / other changes scenario per game launcher. 
1) For each game store, we store a mapping for app id --> store game id. 
2) Update its' launch options when the user does an import for the game
Or is there a better way of doing this?
